Snippets/Core/Core_6/Handlers/AsyncHandler.cs

```diff
@@ -0,0 +1,167 @@
+﻿namespace Core6.Handlers
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Common;
+    using NServiceBus;
+
+    #region ShortComputeBoundMessageHandler
+    public class ShortComputeBoundHandler : IHandleMessages<MyMessage>
+    {
+        public Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            ComputeBoundComponent.BlocksForAFewMilliseconds();
+
+            return Task.FromResult(0); // or Task.CompletedTask
+        }
+    }
+    #endregion
+
+    #region LongComputeBoundMessageHandler
+    public class LongComputeBoundHandler : IHandleMessages<MyMessage>
+    {
+        public Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            return Task.Run(() => ComputeBoundComponent.BlocksForMoreThanHundredMilliseconds());
+        }
+    }
+    #endregion
+
+    public class ComputeBoundComponent
+    {
+        public static void BlocksForAFewMilliseconds()
+        {
+        }
+
+        public static void BlocksForMoreThanHundredMilliseconds()
+        {
+        }
+    }
+
+    #region HandlerReturnsATask
+    public class HandlerReturnsATask : IHandleMessages<MyMessage>
+    {
+        public Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            return SomeLibrary.SomeAsyncMethod(message);
+        }
+    }
+    #endregion
+
+    #region HandlerAwaitsTheTask
+    public class HandlerAwaitsTheTask : IHandleMessages<MyMessage>
+    {
+        public async Task Handle(MyMessage message, IMessageHandlerContext context)
+        {
+            await SomeLibrary.SomeAsyncMethod(message);
```

Comment:
@ramonsmits the point about this example is to show the difference and explain it in the text